packages/mint-components/src/components/sqm-coupon-code/useCouponCode.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+  useProgramId,
+  useQuery,
+  useUserIdentity,
+} from "@saasquatch/component-boilerplate";
+import { useState } from "@saasquatch/universal-hooks";
+import { gql } from "graphql-request";
+import { CopyTextViewProps } from "../views/copy-text-view";
+
+interface CouponCodeProps {
+  programId?: string;
+  tooltiptext: string;
+  tooltiplifespan: number;
+  cancelledErrorText: string;
+  pendingErrorText: string;
+  expiredErrorText: string;
+  redeemedErrorText: string;
+}
+
+type FuelTankReward = {
+  fuelTankCode: string;
+  fuelTankType: string;
+  statuses: string;
+  dateGiven: number;
+  dateExpires: number;
+  dateRedeemed: number;
+  dateCancelled: number;
+  dateScheduledFor: number;
+  datePendingForUnhandledError: number;
+};
+
+export type RewardStatusType =
+  | "AVAILABLE"
+  | "EXPIRED"
+  | "REDEEMED"
+  | "CANCELLED"
+  | "PENDING"
+  | "EMPTY_TANK"
```

Comment:
Should use `kebab-case`

Reply:
I think having these enum types in all caps is more standard to what we do. Will need @Locrian24 to chime in